pkg/k8shandler/logstore.go

```diff
@@ -6,20 +6,88 @@ import (
   "k8s.io/apimachinery/pkg/api/errors"
   "github.com/sirupsen/logrus"
 
+  "bytes"
+  "math/rand"
+  "encoding/base64"
+  "time"
+  "io/ioutil"
+
   metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
   sdk "github.com/operator-framework/operator-sdk/pkg/sdk"
   logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1alpha1"
 )
 
-func CreateOrUpdateLogStore(logging *logging.ClusterLogging)(error) {
-  return nil
+func CreateOrUpdateLogStore(logging *logging.ClusterLogging) error {
+  CreateOrUpdateSecret(logging)
+  return CreateOrUpdateCR(logging)
+}
+
+func getFileContents(filePath string) []byte {
+  contents, err := ioutil.ReadFile(filePath)
+  if err != nil {
+    logrus.Fatalf("Unable to read file to get contents: %v", err)
+  }
+
+  return contents
+}
+
+func init() {
+  rand.Seed(time.Now().UnixNano())
+}
+
+var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
+
+func getRandomWord(wordSize int) string {
+  b := make([]rune, wordSize)
+  for i := range b {
+    b[i] = letters[rand.Intn(len(letters))]
+  }
+  return string(b)
+}
+
+func createPasswdFile(user_name string, user_passwd string) []byte {
+  var buffer bytes.Buffer
+
+  buffer.WriteString(user_name)
+  buffer.WriteString(":\n  passwd: ")
+  buffer.WriteString(base64.StdEncoding.EncodeToString([]byte(user_passwd)))
+
+  return buffer.Bytes()
 }
 
-func CreateOrUpdateSecret(logging *logging.ClusterLogging)(error) {
+func CreateOrUpdateSecret(logging *logging.ClusterLogging) error {
+
+  esSecret := &v1.Secret{
+    TypeMeta: metav1.TypeMeta{
+      Kind: "Secret",
+      APIVersion: "v1",
+    },
+    ObjectMeta: metav1.ObjectMeta{
+      Name: "logging-elasticsearch",
+      Namespace: logging.Namespace,
+    },
+    Type: "Opaque",
+    Data: map[string][]byte{
+      "elasticsearch.key": getFileContents("/tmp/_working_dir/elasticsearch.key"),
+      "elasticsearch.crt": getFileContents("/tmp/_working_dir/elasticsearch.crt"),
+      "logging-es.key": getFileContents("/tmp/_working_dir/logging-es.key"),
+      "logging-es.crt": getFileContents("/tmp/_working_dir/logging-es.crt"),
+      "admin-key": getFileContents("/tmp/_working_dir/system.admin.key"),
+      "admin-cert": getFileContents("/tmp/_working_dir/system.admin.crt"),
+      "admin-ca": getFileContents("/tmp/_working_dir/ca.crt"),
+      "passwd.yml": createPasswdFile("prometheus", getRandomWord(16)),
```

Comment:
@t0ffel how can we get this info to ES?

Reply:
@jcantrill for now I'm going to leave this generation in here and if you're able to update the es plugin to do prometheus auth with a token then we can remove this.

Thoughts?